### PR TITLE
Unified CommentsParser and RobotCommentsParser

### PR DESCRIPTION
**Pull request by EFregnan**
Hi, I merged CommentsParser and RobotCommentsParser, so now CommentsParser is in charge of analyzing both comments and robot comments. 

I noticed that the constructor of ChangeApiRestClient class was using quite a lot of parameters, especially after the new classes I added recently. 
I remember we already somehow discussed this and you were in favour of having all parsers explicitly passed in the constructor. 

However, to somehow reduce the constructor's parameters, I merged two parsers with similar responsibilities. I think this could be a good compromise between clarity and complexity.

What do you think?

### CODE REVIEW

File: src/test/java/com/urswolfer/gerrit/client/rest/http/changes/ChangeApiRestClientTest.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public void testListReviewers() throws Exception {
 
         ChangeApiRestClient changeApiRestClient = new ChangeApiRestClient(gerritRestClient, null, null, null, null,
             null, null, null, null, null,
-            null, null, reviewerInfoParser, null, null, null,
+            null,  reviewerInfoParser, null, null, null,
```
Comment (uwolfer) [inline src/test/java/com/urswolfer/gerrit/client/rest/http/changes/ChangeApiRestClientTest.java]: (remove duplicated whitespace)

File: src/test/java/com/urswolfer/gerrit/client/rest/http/changes/ChangeApiRestClientTest.java [hunk 2]
```diff
@@ -257,7 +257,7 @@ public void testSuggestReviewers() throws Exception {
         EasyMock.expect(suggestedReviewerInfoParser.parseSuggestReviewerInfos(jsonElement)).andReturn(expectedSuggestedReviewerInfos).once();
         EasyMock.replay(suggestedReviewerInfoParser);
 
-        ChangeApiRestClient changeApiRestClient = new ChangeApiRestClient(gerritRestClient, null, null, null,null,
+        ChangeApiRestClient changeApiRestClient = new ChangeApiRestClient(gerritRestClient, null, null,null,
```
Comment (uwolfer) [inline src/test/java/com/urswolfer/gerrit/client/rest/http/changes/ChangeApiRestClientTest.java]: (add missing whitespace)

File: src/test/java/com/urswolfer/gerrit/client/rest/http/changes/ChangeApiRestClientTest.java [hunk 3]
```diff
@@ -449,7 +449,7 @@ public void testGetHashtags() throws Exception {
         EasyMock.expect(hashtagsParser.parseHashtags(jsonElement)).andReturn(expectedHashtags).once();
         EasyMock.replay(hashtagsParser);
 
-        ChangeApiRestClient changeApiRestClient = new ChangeApiRestClient(gerritRestClient, null, null, null, null,
```
Comment (uwolfer) [inline src/test/java/com/urswolfer/gerrit/client/rest/http/changes/ChangeApiRestClientTest.java]: (add missing whitespace)

**Comment by EFregnan** [comment]
Thank you! 
The new commit I pushed should have fixed the white-space issues you noticed, thank you for the review.

**Pull request merged**